queries/gnuplot/highlights.scm

```diff
@@ -0,0 +1,583 @@
+   ]+ @attribute)
+
+(dash_opts "solid" @attribute)
+
+(tics_opts
+  [
+   "axis"
+   "mirror"
+   "inout"
+   ("scale" ["default" ((_) ("," (_))?)]?)
+   ("rotate" "by" @attribute angle: (_))
+   "norotate"
+   [("offset" (position)) "nooffset"]
+   "align"
+   "add"
+   "autofreq"
+   "format"
+   "enhanced"
+   ("format" (_))
+   "log"
+   "rangelimited"
+   ("tc" [(colorspec) ("lt" (_))])
+   ]+ @attribute)
+
+(label_opts
+  [
+   "norotate"@attribute
+   ("rotate" @attribute("by" @attribute degrees: (_))?)
+   "enhanced"@attribute
+   "fb"@attribute
+   ("tc" @field [(colorspec) (["lt" "ls"] @field (_))])
+   ("offset" @attribute(position))
+   "align"@attribute
+   ("at" @attribute (position))
+   "nopoint"@attribute
+   "point"@attribute
+   "boxed"@attribute
+   "nobox"@attribute
+   "hypertext"@attribute
+   ])
+
+(style_opts
+  [
+   ("ls" (_))
+   ("lt" [(_) (colorspec) "black"@attribute "bgnd"@attribute "nodraw"@attribute])
+   ("lw" (_))
+   ("lc" [(_) (colorspec)])
+   "dt"
+   "pt"
+   ("ps" [(_) "variable" @attribute])
+   "pi"
+   "pn"
+   ("as" (_))
+   ("fs" (fill_style))
+   ("fc" [(_) ("lt" @field (_)) ("ls" @field (_))])
+   "nohidden3d"
+   "nocontours"
+   ; "nosurface"
+   "palette"
+   ]+ @field)
+
+(system) @attribute
+
+(macro) @function.macro
+(datablock) @function.macro
+
+(function name: (identifier) @function)
+((function name: (identifier) @function.builtin)
+ (#match? @function.builtin
+  "^(abs|acos|acosh|airy|arg|asin|asinh|atan|atan2|atanh|besj0|besj1|besjn|besy0|besy1|besyn|besi0|besi1|besin|cbrt|ceil|conj|cos|cosh|EllipticK|EllipticE|EllipticPi|erf|erfc|exp|expint|floor|gamma|ibeta|inverf|igamma|imag|int|invnorm|invibeta|invigamma|LambertW|lambertw|lgamma|lnGamma|log|log10|norm|rand|real|round|sgn|sin|sinh|sqrt|SynchrotronF|tan|tanh|uigamma|voigt|zeta|cerf|cdawson|faddeva|erfi|FresnelC|FresnelS|VP|VP_fwhm|Ai|Bi|BesselH1|BesselH2|BesselJ|BesselY|BesselI|BesselK|gprintf|sprintf|strlen|strstrt|substr|strptime|srtftime|system|trim|word|words|time|timecolumn|tm_hour|tm_mday|tm_min|tm_mon|tm_sec|tm_wday|tm_week|tm_yday|tm_year|weekday_iso|weekday_cdc|column|columnhead|exists|hsv2rgb|index|palette|rgbcolor|stringcolumn|valid|value|voxel)$"))
+
+((identifier) @variable.builtin
+  (#match? @variable.builtin
+   "^\\w+_(records|headers|outofrange|invalid|blank|blocks|columns|column_header|index_(min|max)(_x|_y)?|(min|max)(_x|_y)?|mean(_err)?(_x|_y)?|stddev(_err)?(_x|_y)?)$"))
+((identifier)@variable.builtin
+  (#match? @variable.builtin
+   "^\\w+_(sdd(_x|_y)?|(lo|up)_quartile(_x|_y)?|median(_x|_y)?|sum(sq)?(_x|_y)?|skewness(_err)?(_x|_y)?)$"))
+((identifier) @variable.builtin
+  (#match? @variable.builtin
+   "^\\w+_(kurtosis(_err)?(_x|_y)?|adev(_x|_y)?|correlation|slope(_err)?|intercept(_err)?|sumxy|pos(_min|_max)_y|size(_x|_y))$"))
+
+((identifier) @variable.builtin
+  (#match? @variable.builtin "^((GPVAL|MOUSE|FIT)_\\w+|GNUTERM|NaN|VoxelDistance|GridDistance|pi)$"))
```

Comment:
Merging these into one query should be more performant.

Reply:
I had exactly that sometime ago, but if my memory doesn't fail me I had problems with the highlighting and that's why I ended chopping it in parts. I'll try to merge it again but for now it works so I'll leave it as is.